File: app/src/main/java/com/frq/ytmusic/domain/model/Lyrics.kt
package com.frq.ytmusic.domain.model

data class Lyrics(
    val type: LyricsType,
    val lines: List<LyricsLine>,
    val source: String
)

data class LyricsLine(
    val text: String,
    val startTimeMs: Long?,
    val endTimeMs: Long?
)

enum class LyricsType {
    SYNCED,
    PLAIN
}

File: backend/app/models/lyrics.py
"""
Lyrics-related models for structured lyrics response.
Supports both plain text and synced (timed) lyrics.
"""
from typing import List, Optional
from pydantic import BaseModel


class LyricsLine(BaseModel):
    """A single line of lyrics with optional timing information."""
    text: str
    start_time_ms: Optional[int] = None
    end_time_ms: Optional[int] = None


class LyricsData(BaseModel):
    """
    Structured lyrics response.
    
    Attributes:
        type: "synced" if lyrics have timing, "plain" if not
        lines: List of lyrics lines
        source: Source of lyrics (e.g., "youtube_music")
    """
    type: str  # "synced" | "plain"
    lines: List[LyricsLine]
    source: str = "youtube_music"


class LyricsResponse(BaseModel):
    """Response model for lyrics endpoint."""
    video_id: str
    has_lyrics: bool
    lyrics: Optional[LyricsData] = None
